From eaf521f7d6023e3207fc9edf599cf2da0694fb5e Mon Sep 17 00:00:00 2001
From: "lingyan.wang" <lingyan.wang@ck-telecom.com>
Date: Thu, 2 Jul 2015 09:23:01 +0800
Subject: [PATCH] Can't login youtube when is monkey.

[id]:NONE
[Type]
   BUG/FEATURE/ENHANCEMENT
   bugid if bug or feature or enhanchement

[Products]
   HiCam

[Requirement]
   N/A

[Solution Description]
   N/A

[Other Info]
   N/A

Change-Id: I506fc7eff1ba74a6aae84129d24e42d46fdcb902
---
 .../internal/policy/impl/PhoneWindowManager.java   |    2 +-
 .../HiCam/src/com/hicam/camera/CameraPage.java     |    5 +++++
 .../src/com/hicam/setting/ui/ShareOption.java      |    4 ++++
 3 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/frameworks/base/policy/src/com/android/internal/policy/impl/PhoneWindowManager.java b/frameworks/base/policy/src/com/android/internal/policy/impl/PhoneWindowManager.java
index 3d737a3..c886857 100755
--- a/frameworks/base/policy/src/com/android/internal/policy/impl/PhoneWindowManager.java
+++ b/frameworks/base/policy/src/com/android/internal/policy/impl/PhoneWindowManager.java
@@ -2501,7 +2501,7 @@ public class PhoneWindowManager implements WindowManagerPolicy {
             // Hijack modified menu keys for debugging features
             final int chordBug = KeyEvent.META_SHIFT_ON;
 
-            if(!isHiCam())
+            if(!isHiCam() && (!SystemProperties.getBoolean("ro.monkey", false)))
             {
                 if (down && repeatCount == 0) {
                     mHandler.postDelayed(mKeyVolumeDown, 600);
diff --git a/goldsand/packages/apps/HiCam/src/com/hicam/camera/CameraPage.java b/goldsand/packages/apps/HiCam/src/com/hicam/camera/CameraPage.java
index 6b38b99..027035c 100755
--- a/goldsand/packages/apps/HiCam/src/com/hicam/camera/CameraPage.java
+++ b/goldsand/packages/apps/HiCam/src/com/hicam/camera/CameraPage.java
@@ -46,6 +46,7 @@ import android.os.Message;
 import android.os.PowerManager;
 import android.os.PowerManager.WakeLock;
 import android.os.SystemClock;
+import android.os.SystemProperties;
 import android.preference.PreferenceManager;
 import android.provider.Settings;
 import android.support.v4.view.PagerAdapter;
@@ -2719,6 +2720,10 @@ public class CameraPage extends Page implements SurfaceHolder.Callback,
                 dialog.show();
                 break;
             case R.id.youtube_live:
+                if(SystemProperties.getBoolean("ro.monkey", false)){
+                    mLivePlatformDialog.dismiss();
+                    return ;
+                }
 //                PreferenceManager.getDefaultSharedPreferences(mContext).edit().putInt("plat", 2).commit();
                 if (!mLoginControl.hasAvailableNetwork())
                     return;
diff --git a/goldsand/packages/apps/HiCam/src/com/hicam/setting/ui/ShareOption.java b/goldsand/packages/apps/HiCam/src/com/hicam/setting/ui/ShareOption.java
index 85f73b1..37db03b 100755
--- a/goldsand/packages/apps/HiCam/src/com/hicam/setting/ui/ShareOption.java
+++ b/goldsand/packages/apps/HiCam/src/com/hicam/setting/ui/ShareOption.java
@@ -7,6 +7,7 @@ import android.content.Intent;
 import android.content.SharedPreferences;
 import android.content.SharedPreferences.OnSharedPreferenceChangeListener;
 import android.os.Handler;
+import android.os.SystemProperties;
 import android.view.KeyEvent;
 import android.view.View;
 import android.view.ViewGroup;
@@ -99,6 +100,9 @@ public class ShareOption extends WheelViewSettingsOption implements OnAuthResult
         switch (((ShareWheelDataArrayAdapter) mWheelDataAdapter)
                 .getMainStrId(mSelected)) {
         case R.string.youtube_plat_name:
+            if(SystemProperties.getBoolean("ro.monkey", false)){
+                return ;
+            }
             handlePlatRequestLogin(mAuthView = new YtbAuthDialog(mContext));
             break;
         case R.string.twitter_plat_name:
-- 
1.7.9.5

